apps/communications/contacts/test/marionette/lib/views/contact-details/view.js

```diff
@@ -0,0 +1,66 @@
+'use strict';
+
+/* global module */
+
+var ContactsDetailsAccessor = require('./accessors');
+
+function ContactsDetailsView(client, subject) {
+	this.client = client;
+	this.subject = subject;
+	this.accessors = new ContactsDetailsAccessor(client);
+}
+
+ContactsDetailsView.prototype = {
+	makeFavorited: function() {
+		this.accessors.favoritesButton.tap();
+		return this.waitForFavorited;
+	},
+
+	waitForFavorited: function() {
+		function waitFav(temp) {
+        	return temp.detailsHeader.getAttribute('class'
+        	).indexOf('favorite') != -1;
+      	}
+      	this.client.waitFor(waitFav(this.accessors));
+	},
+
+	backToList: function() {
```

Comment:
Okay to duplicate this function from `contacts.js` now. Nonetheless, I'd leave a TODO comment say that this clean needs to be done by creating a ContactsViewBase class. Then ContactsDetailsView would inherit from it.